### PR TITLE
Fix issue 138: Clear focus from all views by default

### LINKED ISSUE #138: Investigate: Focus is inconsistently applied to Views, causing test flakiness
After running several tests, particularly on higher API levels, you might not some elements appear to be focused or selected. This happens randomly and inconsistently and causes significant flakiness.

Consider a mechanism to force the focus consistently on every test run.

### PR DESCRIPTION
**Pull request by DanielJette**
Resolves #138 

## What this PR does

I have observed several tests exhibiting flaky behaviour due to intermittent application of ["focus"](https://blog.novoda.com/do-you-even-focus-bro/) to `Views`. Depending on how the user has interacted with the emulator, you can place it in a state where Android attempts to place focus on your Activity when testing. On most views, having focus results in a visual change. For instance, a highlight box or change in colour.
As such, this random application of focus to your Activity under test will result in inconsistent, flaky screenshots.

To address this, this PR introduces a `FocusModification` which is applied to each Activity when testing. It temporarily removes the `focusable` attribute from all focus targets and immediately restores this property. Doing so triggers Android to remove the focus from the views.

You can see the impact of this change on the FAB in the updated screenshots.

### CODE REVIEW

**Comment by DanielJette** [comment]
Hmm ... CI is failing on the exact test that it should be passing on now. It's possible this isn't working properly. I'll investigate closer

File: Library/src/main/java/com/shopify/testify/internal/modification/FocusModification.kt [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        get() {
+            return this.currentFocus != null
+        }
+
+    private fun View.removeFocusable() {
+        this.isFocusableInTouchMode = false
+        this.isFocusable = false
+    }
+
+    private fun View.restoreFocusable() {
```
Comment (jaredh) [inline Library/src/main/java/com/shopify/testify/internal/modification/FocusModification.kt:67]: `restore` function name makes me think it's returning a mutated state to it's original value. It's really just `disableFocusable`.

File: Library/src/main/java/com/shopify/testify/internal/modification/FocusModification.kt [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    override fun qualifies(view: View) = false
+
+    private val Activity.isAnyViewWithFocus: Boolean
+        get() {
+            return this.currentFocus != null
+        }
+
+    private fun View.removeFocusable() {
```
Comment (jaredh) [inline Library/src/main/java/com/shopify/testify/internal/modification/FocusModification.kt:62]: This function looks like a duplicate of `View.restoreFocusable`

Comment (DanielJette) [reply Library/src/main/java/com/shopify/testify/internal/modification/FocusModification.kt:62]: Yeah, I think so. Must have been a mistake when rebasing.

**Comment by DanielJette** [comment]
I'm not happy with this change. I'm closing it for now while I rethink the strategy

**Pull request closed without merging**